text/0000-infered-types.md

```diff
@@ -0,0 +1,149 @@
+
+
+my_function(MyStruct2::do_something().into()); // ✅
+
+
+my_function(_::do_something().into()); // ❌ error[E0599]: variant or associated item not found in `MyStruct`
+```
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The underscore operator infers the type of a stuct or enum where there is enough type information in a context in order to infer an exact type. In a statement like `_::Variant`, you can imagine the underscore to be the type. That means that anything you could do with an actual type like `MyEnum::Variant` will still apply in an infered enum. Ultimately, the enum or struct doesn't need to be imported but, traits and other specfied things will need to be imported.
+
+Due to how the rust compiler currently works, lots of changes will need to be made to allow paths to be infered in an order that allows for all of the mentioned. One issue is getting the type of a given context mentioned in [rust-lang/rust#8995](https://github.com/rust-lang/rust/issues/8995).
+
+Finally, here are some examples of non-strict typings that can not be allowed.
+```rust
+fn convert<T>(argument: T) -> Example {/* ... */}
+
+do_something(convert(_::new()))
+//                   ^^^^^^^^ Cannot infer type on generic type argument
+```
+
+However, ones where a generic argument can collapse into strict typing can be allowed. The below works because `T` becomes `Example`. This wouldn’t work if `do_something`, however, took a generic.
+```rust
+fn do_something(argument: Example) {/* ... */}
+fn convert<T>(argument: T) -> T {/* ... */}
+
+do_something(convert(_::new()))
+```
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+In the thread [[IDEA] Implied enum types](https://internals.rust-lang.org/t/idea-implied-enum-types/18349), many people had a few concerns about this feature. 
+
+These RFCs could create bugs. An example of this is if a function changes has two enum parameters that share common variant names. Because it’s implied, it would still compile with this bug createing unintended behavior wharas by specifying the type names, the compiler would thrown an error.
+```rust
+enum RadioState {
+   Disabled,
+   Enabled,
+}
+
+enum WifiConfig {
+   Disabled,
+   Reverse,
+   Enabled,
+}
+
+fn configure_wireless(radio: RadioState, wifi: WifiConfig) { /* ... */ }
+```
+
+Another issue with this is that the syntax `_::` could be mistaken for `::` meaning crate.
+
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+There have been many ideas for what this operator should be including `::` add `.`. Despite that, the underscore is the best because it has already been used to infer lifetimes. Additionally the underscore by itself can be used to construct a struct creating a consistent experience. Maintainers should accept this proposal because it can simplify writing Rust code and prevent the large problem of reputition in switch statements. 
```

Comment:
Underscore is also used to infer generic type parameters.

```suggestion
There have been many ideas for what this operator should be including `::` add `.`. Despite that, the underscore is the best because it has already been used to infer lifetimes and generic types. Additionally the underscore by itself can be used to construct a struct creating a consistent experience. Maintainers should accept this proposal because it can simplify writing Rust code and prevent the large problem of repitition in switch statements. 
```

Reply:
"repetition", not "repitition" 😉